### DISCUSSION TITLE
tutorials/editor/using_the_web_editor

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#18] [User-contributed notes]
# tutorials/editor/using_the_web_editor

Since Godot 3.3, there is a Web editor you can use to work on new or existing projects. Browser support: The Web editor requires support for WebAssembly's SharedArrayBuffer. This is in turn require...

https://docs.godotengine.org/en/latest/tutorials/editor/using_the_web_editor.html

**Comment by ahmed-wael-could** [comment]
extends Sprite


# Declare member variables here. Examples:
# var a = 2
# var b = "text"


# Called when the node enters the scene tree for the first time.
func _ready():
	print("Hello Game Started")


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	print("Game Working")
	print(delta)

**Comment by Anutrix** [comment]
Is Firefox WebGL performance issue note still relevant? Asking since linked bug ticket https://bugzilla.mozilla.org/show_bug.cgi?id=1010527 is closed.

**Reply by Calinou** [reply]
In my experience, yes, albeit not as much as before. You'll still get better performance in Chromium-based browsers for WebGL apps, even on Windows/macOS.

**Comment by yoyomule** [comment]
Can a web editor be self-deployed onto my own server, or is it only usable within the official environment provided?

**Reply by Calinou** [reply]
You can self-host the web editor files by compiling it from source, then deploying the files on a web server with cross-origin isolation headers configured (required for SharedArrayBuffer to work): https://docs.godotengine.org/en/stable/contributing/development/compiling/compiling_for_web.html#building-the-editor

**Reply by akien-mga** [reply]
You don't even need to compile from source, every Godot release includes a zip with the web editor sources.

See the `web_editor.zip` file in e.g. the latest stable release: https://github.com/godotengine/godot-builds/releases/latest

**Comment by fariazz** [comment]
Is there a way to have a pre-loaded project in the web editor? For instance we wanna use it for education, and it'd be nice to have say a pre-loaded project with a challenge, or with starter assets.

**Reply by Calinou** [reply]
On the web editor's landing page, you can see a **Preload project ZIP** field. This field doesn't have any default value, and `<input type="file">` in HTML can't have a default value set. Even if this was possible, this field could only refer to a file on the user's local filesystem, not a remote file the user hasn't downloaded yet.

This would require significant development to bundle a ZIP archive on an external URL and have the Godot web editor download and extract it once the project manager starts: https://github.com/godotengine/godot-proposals/issues/6996

**Reply by fariazz** [reply]
Thanks @Calinou that makes a lot of sense. I'll keep an eye out for that proposal in case it is ever developed. 

The use case is essentially education, but people might even use it for bug reports, similar to how people use Codepen and similar.

**Reply by bezark** [reply]
Speaking as a Godot educator, this would be a hugely helpful feature! Is there a way to get involved in/a good entry point for making this happen?

**Reply by Calinou** [reply]
@bezark I suggest opening a dedicated proposal for this: https://github.com/godotengine/godot-proposals

**Comment by michael714** [comment]
Hey Folks, I am trying to use the Web Editor with the demo file, but there are some problems.  When I try to use your page with the latest version, I cannot load the demo.zip that you provide (I downloaded it and then selected the download to preload).  So, I went back to the most recent stable build of the Web Editor (3.4.1) and I was able to preload and save the project, but the project is automatically upgraded to version 4 and so the 3.4.1 editor cannot edit the project.   So, I went back to the web page for the latest build of the Web Editor  (https://editor.godotengine.org/releases/latest/).  Fortunately, the demo folder I created was still there even though I'm now using the latest version of the Web Editor, so I was able to load the demo project and work with it.

I am writing all of this to say that it would be nice if somebody could look into this and make it so that the demo.zip file just worked with the latest version of the Web Editor.  There is some kind of problem there.  I do have screenshots if anyone is interested.

**Comment by dragon20161027** [comment]
in the web editor im start my projekt and the projekt loading forever

**Reply by Calinou** [reply]
Please open an issue on https://github.com/godotengine/godot/issues while including browser and system information.
